SQL_module/SubjectForm_Task2/db_manipulation.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+    include "./db_connection.php";
+
+    $table_name = "electives";
+
+    function insertInTable($title, $description, $lecturer){
+        try {
+
+            $connection = db_connection();
+            $sql = "INSERT INTO electives(title,description,lecturer) VALUES (:title, :description, :lecturer)";
+    
+            $preparedSql = $connection->prepare($sql) or die("Failed to prepare sql query.");
+            $preparedSql->bindParam(':title', $title);
+            $preparedSql->bindParam(':description', $description);
+            $preparedSql->bindParam(':lecturer', $lecturer);
+            
+            $preparedSql->execute() or die("Filed to execute sql query.");   
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
+        }
+    }
+
+    function printContentOfTable(){
+        
+        global $table_name;
+        try {
+            $connection = db_connection();
+            $sql = "SELECT * FROM $table_name";
+            
+            $result = $connection->prepare($sql) or die("Filed to prepare sql query.");
+            $result->execute() or die("Filed to execute sql query.");
+            while($row = $result->fetch(PDO::FETCH_ASSOC)) {
+                echo "ID: " . $row["id"] . "<br/>Title: " . $row["title"] . "<br/>Description: " . $row["description"] . "<br/>Lecturer: " . $row["lecturer"] . "<br/>Created_at: " . $row["created_at"] . "<br/><br/>";
+            }
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
+        }
+    }
+
+    function printSubject($id){
+        global $table_name;
+        try {
+            $connection = db_connection();
+            $sql = "SELECT * FROM $table_name WHERE id = :id;";
+            
+            $result = $connection->prepare($sql) or die("Filed to prepare sql query.");
+            $result->bindParam(':id', $id);
+            $result->execute() or die("Filed to execute sql query.");
+            while($row = $result->fetch(PDO::FETCH_ASSOC)) {
+                echo "Subject ID: " . $row["id"] . "<br/>Title: " . $row["title"] . "<br/>Description: " . $row["description"] . "<br/>Lecturer: " . $row["lecturer"] . "<br/><br/>";
+            }
+        }
+        catch(PDOException $error) {
+            echo $error->getMessage();
+        }
+    }
+
+    function checkIfIDExist($id){
+
+        global $table_name;
+
+        $connection = db_connection();
+        $sql = "SELECT * FROM $table_name WHERE id = :id";
```

Comment:
It would be better to use count() query instead of *. This way it will use indexes and have lower load on the database. At the end you could check just if count() is > 0

Reply:
In the current situation it is actually ok, because you are filtering by id, and the load will be ok.